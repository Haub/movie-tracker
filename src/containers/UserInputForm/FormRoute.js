import React from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';

import SignUpUser from '../../containers/UserInputForm/SignUpUser';
import LogInUser from '../../containers/UserInputForm/LogInUser';
import LogOutUser from '../../containers/UserInputForm/LogOutUser';

export const FormRoute = () => {
  return (
    <Switch>
      <Route exact path="/" component={LogInUser} />
      <Route exact path="/signup" component={SignUpUser} />
      <Route exact path="/:user_id" component={LogOutUser} />
    </Switch>
  )
}

export default withRouter(FormRoute);