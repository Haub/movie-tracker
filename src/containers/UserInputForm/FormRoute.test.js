import React from "react";
import { MemoryRouter } from "react-router";
import { shallow, render } from "enzyme";

import { FormRoute } from "./FormRoute";

describe('FormRoute component', () => {

  it('should match snapshot with default path', () => {
    let wrapper = shallow(<FormRoute store={{}} />);
    render(
      <MemoryRouter initialEntries={['/']}>
        <FormRoute store={{}} />
      </MemoryRouter>
    ); 
    expect(wrapper).toMatchSnapshot();
  });
});