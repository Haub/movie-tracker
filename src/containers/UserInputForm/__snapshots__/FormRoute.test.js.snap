// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRoute component should match snapshot with /login path 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/login"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/signup"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/:user_id"
  />
</Switch>
`;

exports[`FormRoute component should match snapshot with default path 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    component={[Function]}
    exact={true}
    path="/login"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/signup"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/:user_id"
  />
</Switch>
`;
